clients/mssql/store.go

```diff
@@ -0,0 +1,91 @@
+package mssql
+
+import (
+	"strings"
+
+	"github.com/artie-labs/transfer/clients/shared"
+	"github.com/artie-labs/transfer/lib/config"
+	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/db"
+	"github.com/artie-labs/transfer/lib/destination/types"
+	"github.com/artie-labs/transfer/lib/optimization"
+	"github.com/artie-labs/transfer/lib/ptr"
+	_ "github.com/microsoft/go-mssqldb"
+)
+
+type Store struct {
+	configMap *types.DwhToTablesConfigMap
+	config    config.Config
+	db.Store
+}
+
+func getSchema(schema string) string {
+	// MSSQL has their default schema called `dbo`, `public` is a reserved keyword.
+	if strings.ToLower(schema) == "public" {
+		return "dbo"
+	}
+
+	return schema
+}
+
+func (s *Store) Schema(tableData *optimization.TableData) string {
+	return getSchema(tableData.TopicConfig.Schema)
+}
+
+func (s *Store) Label() constants.DestinationKind {
+	return constants.MSSQL
+}
+
+func (s *Store) Merge(tableData *optimization.TableData) error {
+	return shared.Merge(s, tableData, s.config, types.MergeOpts{})
+}
+
+func (s *Store) Append(tableData *optimization.TableData) error {
+	return shared.Append(s, tableData, s.config, types.AppendOpts{
+		TempTableName: s.ToFullyQualifiedName(tableData, true),
+	})
+}
+
+func (s *Store) ToFullyQualifiedName(tableData *optimization.TableData, escape bool) string {
+	return tableData.ToFqName(s.Label(), escape, s.config.SharedDestinationConfig.UppercaseEscapedNames, optimization.FqNameOpts{
+		MsSQLSchemaOverride: s.Schema(tableData),
+	})
+}
+
+func (s *Store) GetTableConfig(tableData *optimization.TableData) (*types.DwhTableConfig, error) {
+	// TODO: Figure out how to leave a comment.
+	const (
+		describeNameCol        = "column_name"
+		describeTypeCol        = "data_type"
+		describeDescriptionCol = "description"
+	)
+
+	return shared.GetTableConfig(shared.GetTableCfgArgs{
+		Dwh:                s,
+		FqName:             s.ToFullyQualifiedName(tableData, true),
+		ConfigMap:          s.configMap,
+		Query:              describeTableQuery(s.Schema(tableData), tableData.RawName()),
+		ColumnNameLabel:    describeNameCol,
+		ColumnTypeLabel:    describeTypeCol,
+		ColumnDescLabel:    describeDescriptionCol,
+		EmptyCommentValue:  ptr.ToString("<nil>"),
+		DropDeletedColumns: tableData.TopicConfig.DropDeletedColumns,
+	})
+}
+
+func LoadStore(cfg config.Config, _store *db.Store) *Store {
+	if _store != nil {
```

Comment:
Done for MSSQL

Reply:
Pulled this out into a separate PR https://github.com/artie-labs/transfer/pull/357

Reply:
We can't move this to tests because shared pkgs like the `ddl` library are loading Redshift and Snowflake directly.

By removing `store` as a 2nd arg, tests are directly going into `db.Open(...)` which will ping the connection and panic if failure occurs.